Contouring.cc

```diff
@@ -151,7 +160,12 @@ void TraceLevel(const float* image, int64_t width, int64_t height, double scale,
 
     // Search RightEdge
     for (j = 0; j < height - 1; j++) {
-        if (image[(j)*width + i] < level && level <= image[(j + 1) * width + i]) {
+        float pt_a = image[(j)*width + i];
+        float pt_b = image[(j + 1) * width + i];
+        pt_a = isnan(pt_a) ? -std::numeric_limits<float>::max() : pt_a;
```

Comment:
Why not just test for nan rather than reassign these values?
if (isnan(pt_a) || isnan(pt_b)) continue;
or
if (!isnan(pt_a) && !isnan(pt_b) && (pt_a > level && level >= pt_b)) {...}

Reply:
simply skipping over `NaN` points will result in an infinite loop, because the `visited` array wouldn't get updated. 

However, you're right that I don't need to reassign these values. I've updated the code with the check 

`if ((isnan(pt_a) || pt_a < level) && level <= pt_b)`, since `level <= `NaN` will always be false, and if `pt_a` is a `NaN`, we still need to trace around it to update the `visited` array.

@confluence @pford please re-review, and @kswang1029 please confirm this still works after commit `26dbbf0`